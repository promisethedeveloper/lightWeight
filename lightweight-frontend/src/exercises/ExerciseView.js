import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import LightWeightAPI from "../api/api";
import LoadingSpinner from "../common/LoadingSpinner";

const ExerciseView = () => {
	const { name } = useParams();

	const [exercise, setExercise] = useState(null);

	useEffect(
		function getExerciseForUser() {
			async function getExercise() {
				try {
					const res = await LightWeightAPI.getExercise(name);
					setExercise(res);
				} catch (error) {
					console.log(error);
				}
			}

			getExercise();
		},
		[name]
	);

	if (!exercise) {
		return <LoadingSpinner />;
	}

	return (
		<div className="d-flex justify-content-end">
			<div className="ExerciseDetail col-md-8 offset-md-2">
				<div className="card" style={{ width: "18rem" }}>
					<img
						src={exercise.gifUrl}
						className="card-img-top"
						alt={exercise.name}
					/>
					<div className="card-body border">
						<h5 className="card-title">{exercise.name}</h5>
						<p className="card-text">
							This exercise targets your {exercise.target}. It is carried out
							using a {exercise.equipment}.
						</p>
					</div>
				</div>
			</div>
		</div>
	);
};

export default ExerciseView;
