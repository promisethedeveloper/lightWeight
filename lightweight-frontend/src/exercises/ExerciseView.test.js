import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ExerciseView from "./ExerciseView";

/** Smoke test */
it("Should render without crashing", () => {
	render(
		<MemoryRouter>
			<ExerciseView />
		</MemoryRouter>
	);
});

/** Snapshot test */
it("Should match snapshots", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<ExerciseView />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("renders the card image element", () => {
	const { container } = render(
		<MemoryRouter>
			<ExerciseView />
		</MemoryRouter>
	);
	const img = container.getElementsByClassName("card-img-top");
	expect(img).toBeTruthy();
});

it("renders the body of the card element", () => {
	const { container } = render(
		<MemoryRouter>
			<ExerciseView />
		</MemoryRouter>
	);
	const cardBody = container.getElementsByClassName("card-body border");
	expect(cardBody).toBeTruthy();
});
