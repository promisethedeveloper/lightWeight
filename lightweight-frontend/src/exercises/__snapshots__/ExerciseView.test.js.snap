// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="LoadingSpinner"
  >
    Loading...
  </div>
</DocumentFragment>
`;
